repose-aggregator/src/docs/asciidoc/filters/keystone-v2.adoc

```diff
@@ -1,52 +1,413 @@
+For example, a service might want all calls authenticated with the exception of the call for WADL retrieval.
+In this situation, you can configure the whitelist as shown in the example below.
+The whitelist contains a list of https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html[Java Regular Expressions] that Repose attempts to match against the full request URI.
+If the URI matches an expression in the white list, then the request is passed to the origin service.
+Otherwise, authentication is performed against the request.
 
 [source,xml]
-.???.cfg.xml
+.keystone-v2.cfg.xml
 ----
-<??? xmlns="http://docs.openrepose.org/repose/???/v1.0">
-    <???> <!--1-->
-    </???>
-</???>
+<?xml version="1.0" encoding="UTF-8"?>
+<keystone-v2 xmlns="http://docs.openrepose.org/repose/keystone-v2/v1.0">
+    <identity-service uri="http://identity.example.com"/>
+    <white-list>
+        <uri-regex>/application\.wadl$</uri-regex> <!--1-->
+    </white-list>
+</keystone-v2>
 ----
-<1> Specifies ???.
+<1> The https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html[Java Regular Expression] to allow matching URI's to pass without requiring authentication.
+
+=== Configuring Cache Timeouts
+This filter caches authentication tokens.
+The length of time that tokens are cached is determined by the Time To Live (TTL) value that is returned from the authentication service (e.g., the Keystone v2 Identity service) during token validation.
+
+You can configure alternate maximum TTL for caching of authentication tokens, groups, and endpoints.
+If you specify the token element value in the configuration file, this value is used when caching tokens, unless the token TTL value provided by the Keystone v2 Identity service is less than the token-cache-timeout value.
+This method prevents Repose from caching stale tokens.
+If the token's TTL exceeds the maximum allowed TTL value (2^31 - 1), the maximum allowed TTL is used.
+
+[source,xml]
+.keystone-v2.cfg.xml
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<keystone-v2 xmlns="http://docs.openrepose.org/repose/keystone-v2/v1.0">
+    <identity-service uri="http://identity.example.com"/>
+    <cache>
+        <timeouts variability="0">     <!--1-->
+            <token>600</token>         <!--2-->
+            <group>600</group>         <!--3-->
+            <endpoints>600</endpoints> <!--4-->
+        </timeouts>
+    </cache>
+</keystone-v2>
+----
+<1> This value will be added or subtracted to the cache timeouts to help ensure that the cached items have some variability so they don't all expire at the exact same time. (Default: 0)
+<2> The number of seconds which cached tokens will live in the datastore.
+<3> The number of seconds which cached groups will live in the datastore.
+<4> The number of seconds which cached endpoints will live in the datastore.
+
+[NOTE]
+====
+Each timeout value behaves in the following way:
+
+* If `-1`, caching is disabled.
+* If `0`, data is cached indefinitely. +
```

Comment:
Yep. That was the intent based on the verbiage earlier, but it does muddy it. Updated.